src/MySqlConnector/Core/ServerSession.cs

```diff
@@ -416,7 +416,7 @@ public async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancella
 			InitialHandshakePayload initialHandshake;
 			do
 			{
-				shouldRetrySsl = (sslProtocols == SslProtocols.None || (sslProtocols & SslProtocols.Tls12) == SslProtocols.Tls12) && Utility.IsWindows();
+				shouldRetrySsl = sslProtocols == SslProtocols.None && Utility.IsWindows();
```

Comment:
This logic is incorrect; we should still retry if the user put `TlsVersion=Tls11,Tls12` in their connection string, for example.

```suggestion
				shouldRetrySsl = (sslProtocols == SslProtocols.None || ((sslProtocols & SslProtocols.Tls12) == SslProtocols.Tls12) && (sslProtocols & (SslProtocols.Tls | SslProtocols.Tls11) != SslProtocols.None)) && Utility.IsWindows();
```

(That's getting a little hard to read, so could be refactored into some temporary variables.)

We'd also want to change line 528 to fall back to the non-1.2 protocols the user had specified (unless they used `None`, in which case 1.0/1.1).